src/main/java/edu/tamu/sage/job/JobRunner.java

```diff
@@ -0,0 +1,98 @@
+package edu.tamu.sage.job;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import edu.tamu.sage.model.Job;
+import edu.tamu.sage.model.repo.JobRepo;
+import edu.tamu.sage.service.ProcessorService;
+
+@Service
+public class JobRunner {
+    @Autowired
+    JobRepo jobRepo;
+
+    @Autowired
+    ProcessorService processorService;
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US);
+
+    private static final String START_TIME_KEY = "startTime";
+    private static final String DATE_KEY = "date";
+    private static final String DAYS_KEY = "days";
+    private static final String MONTHS_KEY = "months";
+
+    @Scheduled(cron = "0 0/30 * * * ?")
+    private void runJobs() {
+        logger.debug("Checking "+jobRepo.findByScheduleActiveTrue().size()+" Active Jobs");
+        LocalDateTime schedulerStarted = java.time.LocalDateTime.now(ZoneId.of("UTC"));
+        jobRepo.findByScheduleActiveTrue().forEach(j -> {
+            switch (j.getSchedule().getFrequency()) {
+                case ONCE:
+                    LocalTime jobStartTime = buildJobStartTime(j.getSchedule().getScheduleData().get(START_TIME_KEY));
+                    LocalDateTime jobDate = LocalDateTime.parse(j.getSchedule().getScheduleData().get(DATE_KEY), getDateTimeFormatter());
+
+                    if (jobDate.getDayOfMonth() == schedulerStarted.getDayOfMonth() && jobStartTime.getHour() == schedulerStarted.getHour() && jobStartTime.getMinute() == schedulerStarted.getMinute()) {
```

Comment:
With cron running every 30 minutes starting from 0 minute on the hour, how could scheduledStarted minutes equal 58 other possible jobStartTime minutes?

Reply:
The UI is read only limited to hour and half hour minutes.